lib/modules/datasource/galaxy-collection/types.ts

```diff
@@ -4,10 +4,16 @@ export interface BaseProjectResult {
   latest_version: {
     version: string;
   };
+  // new in v3, replaces latest_version
```

Comment:
🤔   
I would prefer to split this up instead of using optional parameters.

```typescript
export interface BaseProjectResult {
  versions_url: string;
  deprecated: boolean;
  api_version: string; // if such thing exists
}

export interface BaseProjectResultV2 extends BaseProjectResult{
  latest_version: {
    version: string;
  };
}

export interface BaseProjectResultV3 extends BaseProjectResult{
  highest_version: {
    version: string;
  };
}
```

You can use full type safety like this: 
```typescript

function isBaseProjectResultV2(result: BaseProjectResult): result is BaseProjectResultV2 {
  return !is.nullOrUndefined("latest_version" in result)
}

function isBaseProjectResultV3(result: BaseProjectResult): result is BaseProjectResultV3 {
  return !is.nullOrUndefined("highest_version" in result)
}

function convertBaseProject(base: BaseProjectResult): BaseProjectResultV2 | null {
  if (isBaseProjectResultV2(base)) {
    return base
  }
  if (isBaseProjectResultV3(base)) {
    return {
      ...base,
      latest_version: base.highest_version
    }
  }
  return null
}

// then you can use it like this
    const baseProject = convertBaseProject(galaxyUrlResponse.body);
    if (is.nullOrUndefined(baseProject)) {
      logger.debug("log something")
      return null
    }
```